### ISSUE TITLE
[feature] add fuzzy finding

### ISSUE THREAD

**Issue by rvion** [Globidev/term-picker#2]
you could use

> ### window.getstr([y, x])
> 
> Read a string from the user, with primitive line editing capacity.

from here: http://stackoverflow.com/questions/21784625/how-to-input-a-word-in-ncurses-screen

**Comment by Globidev** [comment]
Yep, already made a working prototype with the levenshtein distance.

Still having issues with how to properly display it
colorize matching letters? auto-expanding matching subitems? selecting best match? showing the current search string in a vim-like status bar?

**Comment by rvion** [comment]
#### experimental suggestion:
- [ ] auto-expand subitem tree, and only show elems with good enough matching value
- [ ] colorize matching letters( yellow ?)
- [ ] showing the current search string as the first line (move down the tree to second line)
- [ ] selecting best match

#### standard behaviour

if it is weird on screen, you could switch back to how sublime/atom/vim way are doing:
- [ ] when user type text to search items, drop the tree view sort a falt list by mathcing score

ex: 

```
query: lem1

tree/folder1/elem1
tree/folder2/elem1
tree/folder3/elem1
tree/folder1/elem11
tree/folder1/elem21
tree/test/yolo
```
